python/ray/serve/tests/test_util.py

```diff
@@ -665,6 +667,47 @@ def test_get_all_live_placement_group_names(ray_instance):
     assert set(get_all_live_placement_group_names()) == {"pg3", "pg4", "pg5", "pg6"}
 
 
+def test_metrics_pusher_basic():
+    counter = {"val": 0}
+
+    def task(c):
+        time.sleep(0.001)
+        c["val"] += 1
+
+    metrics_pusher = MetricsPusher()
+    metrics_pusher.register_task(lambda: task(counter), 0.5)
+    metrics_pusher.start()
+
+    # At 10 seconds, the task should have executed 20 times.
+    # Sleep until just under 10 + 0.5 seconds to give leeway
+    # for accumulated error.
+    time.sleep(10.4)
```

Comment:
this makes my skin crawl a bit -- not only will these tests be very slow, they are also nearly guaranteed to be flaky

instead of relying on actual sleep, please mock or fake `time` and write true unit tests

Reply:
That makes sense, I've mocked out timer and rewrote the tests a bit - lmk if this looks better!